src/hotspot/share/gc/g1/g1HeapVerifier.cpp

```diff
@@ -410,8 +410,8 @@ class VerifyRegionListsClosure : public HeapRegionClosure {
       _old_count++;
     } else {
       // There are no other valid region types. Check for one invalid
-      // one we can identify: pinned without old or humongous set.
-      assert(!hr->is_pinned(), "Heap region %u is pinned but not old or humongous.", hr->hrm_index());
+      // one we can identify before crashing: non-movable.
+      assert(hr->is_young_gc_movable(), "Heap region %u is non-movable.", hr->hrm_index());
```

Comment:
Given this branch is unreachable, can one just use `fatal(...)` without a predicate?

Reply:
> Given this branch is unreachable, can one just use `fatal(...)` without a predicate?

I also thought about this: the purpose of the additional assert is/was (I think) to fail with a better message than just crashing with some nondescript message, filtering out other possible cases. 
Will remove this after all.